gui/src/hw.rs

```diff
@@ -341,60 +468,156 @@ impl HardwareWallets {
         }
     }
 
-    pub fn refresh(&self) -> iced::Subscription<HardwareWalletMessage> {
+    pub fn reset_refresh(&mut self) {
+        // In order to reset the subscription state we need to change the
+        // subscription id
+        self.refresh_index = self.refresh_index.wrapping_add(1);
+    }
+
+    pub fn refresh(&self, taproot: bool) -> iced::Subscription<HardwareWalletMessage> {
+        let id = format!(
+            "refresh-{}-{}-{}",
+            self.network, self.refresh_index, taproot
+        );
         iced::subscription::unfold(
-            format!("refresh-{}", self.network),
+            id,
             State {
                 network: self.network,
                 keys_aliases: self.aliases.clone(),
                 wallet: self.wallet.clone(),
                 connected_supported_hws: Vec::new(),
+                hws_upgrade: Vec::new(),
                 api: None,
                 datadir_path: self.datadir_path.clone(),
+                taproot,
+                hws: Vec::new(),
+                still: Vec::new(),
+                still_upgrade: Vec::new(),
+                sleep: false,
             },
             refresh,
         )
     }
 }
 
-struct State {
+pub struct State {
     network: Network,
     keys_aliases: HashMap<Fingerprint, String>,
     wallet: Option<Arc<Wallet>>,
     connected_supported_hws: Vec<String>,
     api: Option<ledger::HidApi>,
     datadir_path: PathBuf,
+    taproot: bool,
+    hws: Vec<HardwareWallet>,
+    still: Vec<String>,
+    hws_upgrade: Vec<HardwareWallet>,
+    still_upgrade: Vec<String>,
```

Comment:
Previously we did not need the still field and now we have
still and still_upgrade. Can we just pass one unique `still: &mut Vec<String>` that we fill with ids from "ready" device and `upgrading` to the poll_ledger etc
and use it to pass device new connection has they are either ready or in upgrade state ?

I may miss a subtility

Reply:
`still_upgrade` & `hws_upgrade` need NOT to be reset at the end of `refresh()`: `hws` & `still` are need to be reset, they are not stricly needed in the state, but i add them for convenience & readability as we need to pass them to all `poll_xxx()` and `handle_xxx_device()`